temporalio/lib/temporalio/client/interceptor.rb

```diff
@@ -68,6 +68,23 @@ def intercept_client(next_interceptor)
         :rpc_options
       )
 
+      # Input for {Outbound.list_workflows}.
+      ListWorkflowPageInput = Data.define(
+        :query,
+        :next_page_token,
+        :page_size,
+        :rpc_options
+      )
+
+      # Output for {Outbound.list_workflow_page}.
+      ListWorkflowPageOutput = Data.define(:executions, :next_page_token) do
```

Comment:
Can we move this to `Temporalio::Client` since it's returned from client? The interceptor can return that instead of a dedicated output type. Also, I wonder if we need this enumerable or is asking users to access the `executions` field acceptable enough?